packages/slate/src/interfaces/editor.ts

```diff
@@ -1062,8 +1062,12 @@ export const Editor = {
         }
 
         while (true) {
-          // If there's no more string, continue to the next block.
-          if (string === '') {
+          // If there's no more string (and we've exhausted the available space
+          // or we're at the end of the range), break from the current node.
+          if (
+            string === '' &&
+            (available === 0 || Point.equals(last, { path, offset }))
```

Comment:
Additional constraints for when to break early while scanning text content for possible cursor positions.

Either the available space has been exhausted or we've reached the end of the permitted range. These seem to accommodate existing use cases, while also fixing the bug.